CoastalResilienceNetwork/GeositeFramework
# Setup templating engine and inject region.json into app

## Overview

The next step in the static site conversion is that we add a templating engine that subs in data from custom JSON into templates that comprise the app. I added Jinja2 as the templating engine and process `region.json` for `index.html` as proof of concept that Jinja2 works for our needs. While we don't use Jinja2 frequently, its template syntax is familiar as it is strongly inspired by Django. Jinja2 is very lightweight as it only provides templating functionality and is very popular among pythonistas.

Connects #1086 

### Demo

Serving the templated `index.html` shows that variables are substituted.
<img width="669" alt="screen shot 2018-09-18 at 11 33 52 am" src="https://user-images.githubusercontent.com/10568752/45699152-84c68c00-bb37-11e8-992f-0c4258f97b88.png">

Running the server looks like
<img width="790" alt="screen shot 2018-09-18 at 11 40 14 am" src="https://user-images.githubusercontent.com/10568752/45699258-a4f64b00-bb37-11e8-9ec1-963a63283183.png">

### Notes

There's a lot of context in the commit messages, check them out.
The script `main.py` writes the jinja template to disk. This workflow works for our dev envir and `build.py` in the geosite-build process whereby we will return zipped static site files to the client to serve on their own.

## Testing Instructions

Create the container, install deps, and run the server
`./scripts/update`
`./scripts/server`
Visit http://localhost:54633 and see that variables are slotted in from region.json. The most prominent one is the text "Geosite Framework Sample" in the middle
